LiteCore/Query/SQLiteFleeceFunctions.cc

```diff
@@ -103,6 +107,77 @@ namespace litecore {
         }
     }
 
+    static void handle_fts_value(const Value* data, alloc_slice& result) {
+        switch(data->type()) {
+            case kBoolean:
+                result.append(data->asBool() ? "T"_sl : "F"_sl);
+                break;
+            case kNumber:
+            {
+                string resultStr;
+                if(data->isInteger()) {
+                    if(data->isUnsigned()) {
+                        resultStr = to_string(data->asUnsigned());
+                    } else {
+                        resultStr = to_string(data->asInt());
+                    }
+                } else {
+                    stringstream ss;
+                    if(data->isDouble()) {
+                        ss << data->asDouble();
+                    } else {
+                        ss << data->asFloat();
+                    }
+
+                    resultStr = ss.str();
+                }
+
+                result.append(slice(resultStr));
+                break;
+            }
+            case kString:
+                result.append(data->asString());
+                break;
+            case kArray:
+                handle_fts_array(data, result);
+                break;
+            case kDict:
+                handle_fts_dict(data, result);
+                break;
+        }
+    }
+
+    static void handle_fts_array(const Value* data, alloc_slice& result) {
```

Comment:
Consider using a DeepIterator to walk through the Fleece. It'll make the code simpler, and avoid using unbounded stack space too. (Use the one in DeepIterator.hh, _not_ Fleece.hh.)